### PR TITLE
Policy violation context for Ansible Automation Platform

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/stolostron/backlog/issues/26870

### CODE REVIEW

File: applications/ansible_intro.adoc [hunk 1]
```diff
@@ -39,6 +39,8 @@ With a valid prehook `AnsibleJob`, the subscription launches the prehook `Ansibl
 
 For example, you can have a prehook `AnsibleJob` that failed to propagate a placement rule subscription. When the placement rule decision changes, new prehook and posthook `AnsibleJob` instances are created.
 
+//Based on issue #26870 what are other thoughts about adding a section for GRC Ansible extra variables?
```
Comment (dockerymick) [inline applications/ansible_intro.adoc]: I am on the side to keep the info in the governance folder

Comment (swopebe) [reply applications/ansible_intro.adoc]: I am not sure I understand this comment?

File: applications/ansible_config.adoc [hunk 2]
```diff
@@ -148,4 +148,4 @@ spec:
     version: 1.23.45
 ----
 
-See [Workflows](https://docs.ansible.com/ansible-tower/latest/html/userguide/workflows.html) to learn more about Ansible Workflow.
\ No newline at end of file
+See link:https://docs.ansible.com/ansible-tower/latest/html/userguide/workflows.html[Workflows] to learn more about Ansible Workflow.
```
Comment (swopebe) [inline applications/ansible_config.adoc:151]: thank you!

File: applications/ansible_intro.adoc (hunk 1 above)
Comment (swopebe) [reply applications/ansible_intro.adoc]: Posted in Slack: 

Ansible note from our recent meeting: for some reason the cluster doc was linking to the application doc for Ansible, which should have been written more independently given that the cluster user has not hit the application process necessarily. (That is why I am asking for product-wide awareness on things that touch all components, but the architecture is off right now and we don't have time to redo all Ansible.
So I think we have to do what makes sense with what we have for Ansible and address this in 2.8?? (We probably should have come to the table to announce a need to an Ansible integration section...app was the first one to have it in the doc)

Comment (dockerymick) [reply applications/ansible_intro.adoc]: @swopebe basically, do we want to add the Governance automation configuration section in this file. As you mentioned in your comment, let's revisit the architecture of the Ansible docs as a team. I will not add the info to the Applications folder.

**Comment by dockerymick** [comment]
Related issue: https://github.com/stolostron/backlog/issues/26871

CHANGE FOR 2.6

After this PR is merged, I will move the "concept" info (descriptions of automation options) to the Manage security policies topic. It will be listed in the (new) Governance automation configuration section

**Review by swopebe — APPROVED** [review]

File: governance/ansible_grc.adoc [hunk 3]
```diff
@@ -31,19 +31,14 @@ Configure an automation for a specific policy by clicking *Configure* in the _Au
 
 After a credential is selected, click the Ansible job drop-down list to select a job template. In the _Extra variables_ section, add the parameter values from the `extra_vars` section of the `PolicyAutomation`. Select the frequency of the automation. You can select _Run once mode_, _Run everyEvent mode_, or _Disable automation_.
 
-** _Manual run_: Manually set this automation to run once. After the automation runs, it is set to `disabled`. *Note:* You can only select _Manual run_ mode when the schedule frequency is disabled.
```
Comment (mprahl) [inline governance/ansible_grc.adoc:34]: Why was this removed?

File: governance/manage_policy_intro.adoc [hunk 4]
```diff
@@ -8,35 +8,49 @@ Use the _Governance_ dashboard to create, view, and manage your security policie
...
 
+[#grc-automation-configuration]
+== Governance automation configuration
+
+If there is a configured automation for a specific policy, you can select the automation to view more details. View the following descriptions of the schedule frequency options for your automation:
+
+- _Manual run_: Manually set this automation to run once. After the automation runs, it is set to `disabled`. *Note:* You can only select _Manual run_ mode when the schedule frequency is disabled.
+- _Run once mode_: When a policy is violated, the automation runs one time. After the automation runs, it is set to `disabled`. After the automation is set to `disabled`, you must continue to run the automation manually. When you run _once mode_, the extra variable of `target_clusters` is automatically supplied with the list of clusters that violated the policy. The {aap-short} Job template must have `PROMPT ON LAUNCH` enabled for the `EXTRA VARIABLES` section (also known as `extra_vars`).
++
+** Use the following parameters in the `extra_vars` section to receive additional policy violation details:
```
Comment (mprahl) [inline governance/manage_policy_intro.adoc]: The feature actually provides these variables in `extra_vars` automatically.

File: governance/manage_policy_intro.adoc [hunk 5]
```diff
@@ -8,35 +8,49 @@ Use the _Governance_ dashboard to create, view, and manage your security policie
...
++
+* `policy_name`: The name of the non-compliant root policy that intiates the {aap-short} job on the hub cluster.
+* `namespace`: The namespace of the root policy. 
+* `hub_cluster`: The name of the hub cluster, which is received from the Kubernetes configuration host value.
```
Comment (mprahl) [inline governance/manage_policy_intro.adoc]:
```suggestion
* `hub_cluster`: The name of the hub cluster, which is determined by the value in the `clusters` `DNS` object.
```

File: governance/manage_policy_intro.adoc [hunk 6]
```diff
@@ -8,35 +8,49 @@ Use the _Governance_ dashboard to create, view, and manage your security policie
...
+* `policy_set`: This parameter contains all associated policy set names of the root policy. If the policy is not within a policy set, the `policy_set` parameter is empty.
+* `policy_violation_context`: This parameter contains a list of non-compliant cluster names, and the value is the raw replicated policy status for each non-compliant cluster. The policy automation checks the hub cluster namespaces for replicated policies, then filters the non-compliant, replicated policies.
```
Comment (mprahl) [inline governance/manage_policy_intro.adoc]: There was too many technical details.
```suggestion
* `policy_violation_context`: This parameter contains a list of non-compliant cluster names, and the value is the policy `status` field for each non-compliant cluster.
```

File: governance/manage_policy_intro.adoc (hunk 4 above)
Comment (mprahl) [reply governance/manage_policy_intro.adoc]: These values are also not specific to run once mode. The other modes also have this information.

File: governance/ansible_grc.adoc (hunk 3 above)
Comment (dockerymick) [reply governance/ansible_grc.adoc:34]: Because I moved the info to the manage_policy_intro.adoc. The ansible_grc.adoc is task-oriented. Should the note be added back?

File: governance/manage_policy_intro.adoc [hunk 7]
```diff
@@ -8,35 +8,48 @@ Use the _Governance_ dashboard to create, view, and manage your security policie
...
 
+[#grc-automation-configuration]
+== Governance automation configuration
+
+If there is a configured automation for a specific policy, you can select the automation to view more details. View the following descriptions of the schedule frequency options for your automation:
+
+- _Manual run_: Manually set this automation to run once. After the automation runs, it is set to `disabled`. *Note:* You can only select _Manual run_ mode when the schedule frequency is disabled.
+- _Run once mode_: When a policy is violated, the automation runs one time. After the automation runs, it is set to `disabled`. After the automation is set to `disabled`, you must continue to run the automation manually. When you run _once mode_, the extra variable of `target_clusters` is automatically supplied with the list of clusters that violated the policy. The {aap-short} Job template must have `PROMPT ON LAUNCH` enabled for the `EXTRA VARIABLES` section (also known as `extra_vars`).
+- View the following parameters listed in the `extra_vars` section:
```
Comment (mprahl) [inline governance/manage_policy_intro.adoc]: Could you please move this to below all the modes?

Comment (mprahl) [inline governance/manage_policy_intro.adoc]:
```suggestion
- The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:
```

**Comment by dockerymick** [comment]
@mprahl thanks so much Matt for your review and suggestions. I have completed the updates. Let me know if anything else needs to be updated.

File: governance/manage_policy_intro.adoc [hunk 8]
```diff
@@ -8,35 +8,46 @@ Use the _Governance_ dashboard to create, view, and manage your security policie
...
+- _Run everyEvent mode_: When a policy is violated, the automation runs every time for each unique policy violation per managed cluster. Use the `DelayAfterRunSeconds` parameter to set the minimum seconds before an automation can be restarted on the same cluster. If the policy is violated multiple times during the delay period and kept in the violated state, the automation runs one time after the delay period. The default is 0 seconds and is only applicable for the `everyEvent` mode. When you run `everyEvent` mode, the extra variable of `target_clusters` and {aap-short} Job template is the same as _once mode_.
+- _Disable automation_: When the scheduled automation is set to `disabled`, the automation does not run until the setting is updated.
+- The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:
```
Comment (mprahl) [inline governance/manage_policy_intro.adoc]:
```suggestion

The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:
```

**Review by mprahl — APPROVED** [review]

**Comment by dockerymick** [comment]
Fixed a couple of links. this pr can be merged

File: governance/manage_policy_intro.adoc [hunk 9]
```diff
@@ -8,35 +8,47 @@ Use the _Governance_ dashboard to create, view, and manage your security policie
...
+- _Run everyEvent mode_: When a policy is violated, the automation runs every time for each unique policy violation per managed cluster. Use the `DelayAfterRunSeconds` parameter to set the minimum seconds before an automation can be restarted on the same cluster. If the policy is violated multiple times during the delay period and kept in the violated state, the automation runs one time after the delay period. The default is 0 seconds and is only applicable for the `everyEvent` mode. When you run `everyEvent` mode, the extra variable of `target_clusters` and {aap-short} Job template is the same as _once mode_.
+- _Disable automation_: When the scheduled automation is set to `disabled`, the automation does not run until the setting is updated.
+
+The following variables are automatically provided in the `extra_vars` of the {aap-short} Job:
+
+* `policy_name`: The name of the non-compliant root policy that intiates the {aap-short} job on the hub cluster.
```
Comment (mprahl) [inline governance/manage_policy_intro.adoc:46]: @dockerymick I just noticed a typo here:
`intiates` => `initiates`
